src/main/java/view/leaderboard/LeaderboardView.java

```diff
@@ -206,4 +221,15 @@ public void setLeaderboardController(LeaderboardController leaderboardController
         // Set the controller for the leaderboard view
         this.leaderboardController = leaderboardController;
     }
+
+    public void setLeaderboardChartController(interface_adapter.leaderboard_as_chart.LeaderboardAsChartController controller) {
+        this.leaderboardChartController = controller;
+    }
+
+    private void switchToLeaderboardChart() throws DataAccessException {
+        leaderboardChartController.requestChart();
```

Comment:
```suggestion
        leaderboardChartController.requestChart(50);
```
Now the method `requestChart` requires a parameter `leaderboardEntryCount`.